### PR TITLE
Migrate nitrokey-firmware into monorepo

### LINKED ISSUE #70: Migrate `nitrokey-firmware`
https://github.com/ngi-nix/nitrokey-firmware

Tracking issue: https://github.com/ngi-nix/ngipkgs/issues/12.

### PR DESCRIPTION
**Pull request by matusf**
Closes #70

Tracking issue: https://github.com/ngi-nix/ngipkgs/issues/12.

### CODE REVIEW

File: pkgs/nitrokey-firmware/flake. [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (matusf) [inline pkgs/nitrokey-firmware/flake.:1]: `rm`

File: .github/workflows/pull_request.yaml [hunk 2]
```diff
@@ -15,4 +15,6 @@ jobs:
       - uses: DeterminateSystems/magic-nix-cache-action@main
         with:
           upstream-cache: https://ngi.cachix.org/
-      - run: nix flake check
+      - run: nix flake check --impure
```
Comment (lorenzleutgeb) [inline .github/workflows/pull_request.yaml:18]: Don't do impure if all you want is some unfree packages. Please use `config.allowUnfreePredicate` (via `flake.nix`). Introduce a set of unfree packages, and for each unfree package, leave a comment justifying it.

File: .github/workflows/push.yaml [hunk 3]
```diff
@@ -16,4 +16,6 @@ jobs:
         with:
           name: ngi
           authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
-      - run: nix flake check
+      - run: nix flake check --impure
```
Comment (lorenzleutgeb) [inline .github/workflows/push.yaml:19]: See comment on https://github.com/ngi-nix/ngipkgs/pull/90/files#diff-7b2a967de7060a238e49a57908f83e860b8cb53a0b0991c3d0d275857c1014d4R18

**Review by lorenzleutgeb — CHANGES_REQUESTED** [review]

**Comment by mightyiam** [comment]
I feel I made some good progress. Plan to continue tomorrow.

File: pkgs/default.nix [hunk 4]
```diff
@@ -1,7 +1,19 @@
 {
   lib,
   callPackage,
+  inputs,
 }: let
+  linuxSystem = "x86_64-linux";
```
Comment (mightyiam) [inline pkgs/default.nix:6]: @lorenzleutgeb I'm not sure what to do with this. I'm quite sure I don't understand the overall structure of the flake and that's why I'm confused. I know it's documented with some comments, but I still don't get it. Perhaps I'll take another look. There's just too much passing of values around and I lose track.

File: .github/workflows/push.yaml (hunk 3 above)
Comment (mightyiam) [reply .github/workflows/push.yaml]: The package with unfree deps has since been removed by us.

Comment (mightyiam) [reply .github/workflows/push.yaml]: Consider resolving.

File: .github/workflows/pull_request.yaml (hunk 2 above)
Comment (mightyiam) [reply .github/workflows/pull_request.yaml]: Consider resolving. The package with unfree deps is not present.

**Comment by wamirez** [comment]
Superseded by NixOS/nixpkgs#337959

**Pull request closed without merging**